core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/InstalledIntelliJ.scala

```diff
@@ -219,5 +219,26 @@ final class DownloadedIntelliJ(
   override val ideaProperties: Path =
     root.resolve("bin").resolve("idea.properties").write(ideaPropertiesContent)
 
-  override def cleanup(): Unit = root.delete()
+  override def cleanup(): Unit =
+    probePaths.logExport match {
+      case Some(path) =>
+        root
+          .resolve("logs")
+          .copyDir(path.resolve(getPathWithVersionNumber(root).getFileName).resolve("logs"))
+        root.delete()
+      case None =>
+        root.delete()
```

Comment:
if you always call root.delete() you can move it outside this pattern matching to not repeat it. probePaths.logExport.foreach  { path =>
  // do the copy
}